### PR TITLE
Add dictionary support

### LINKED ISSUE #483: [v1.4] Support user-dictionary loading
*Following this [central issue](https://github.com/meilisearch/integration-guides/issues/286)*

- [ ]  Add a new setting: `dictionary` with `get`, `update`, and `reset` methods associated.
Here are the JS equivalents in meilisearch-js you should create for this repository:

```jsx
client.index('indexName').getDictionary(); // calls GET /indexes/:uid/settings/dictionary
client.index('indexName').updateDictionary(['|', '/', '&sep']);  // calls PUT /indexes/:uid/settings/dictionary
client.index('indexName').resetDictionary(); // calls DELETE /indexes/:uid/settings/dictionary
```

- [ ]  The methods associated with the `/settings` API route (to get, reset and update the settings globally) must be able to receive in the payload the new configuration `dictionary`.
- [ ]  ⚠️ Add integration tests ⚠️
- [ ]  Update the `.code-samples.meilisearch.yaml`
    - [ ]  Add `get_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1133
    - [ ]  Add `update_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1136
    - [ ]  Add `reset_dictionary_1` key and “translate” the following curl example by using the newly added methods: https://github.com/meilisearch/documentation/blob/2060089a5c10bcd0beb28633062f90cf0b279f7f/.code-samples.meilisearch.yaml#L1144

### PR DESCRIPTION
**Pull request by danFbach**
# Pull Request

## Related issue
Fixes #483

## What does this PR do?
- Adds dictionary support
- tests added and passing
- code samples added

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -782,3 +782,10 @@ distinct_attribute_guide_distinct_parameter_1: |-
     Distinct = "sku"
   };
   await client.Index("products").SearchAsync<Product>("white shirt", params);
+get_dictionary_1: |-
+  var indexDictionary = await client.Index("products").GetDictionaryAsync();
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml]:
```suggestion
  var indexDictionary = await client.Index("books").GetDictionaryAsync();
```

File: .code-samples.meilisearch.yaml [hunk 2]
```diff
@@ -782,3 +782,10 @@ distinct_attribute_guide_distinct_parameter_1: |-
...
+update_dictionary_1: |-
+  var newDictionary = new string[] { "J. R. R.", "W. E. B." };
+  await client.Index("products").UpdateDictionaryAsync(newDictionary);
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("books").UpdateDictionaryAsync(newDictionary);
```

File: .code-samples.meilisearch.yaml [hunk 3]
```diff
@@ -782,3 +782,10 @@ distinct_attribute_guide_distinct_parameter_1: |-
...
+reset_dictionary_1: |-
+  await client.Index("products").ResetDictionaryAsync();
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml]:
```suggestion
  await client.Index("books").ResetDictionaryAsync();
```

**Comment by curquiza** [comment]
bors try

**Review by curquiza — CHANGES_REQUESTED** [review]

**Comment by curquiza** [comment]
@danFbach 
looks like the tests do not pass with you PR. Can you fix the tests?

**Comment by danFbach** [comment]
okay, i see the issue. I needed to assign the Dictionary in `SettingsWithDefaultedNullFields`.
sending commit now.

**Review by curquiza — APPROVED** [review]
Thank you again @danFbach 

bors merge

**Pull request merged**